html/controls/sk/sk-sheet.js

```diff
@@ -154,48 +142,95 @@ function prepareSkSheetTable(element, teamId, mode) {
+				$('<td>').addClass('SmallHead').append($('<div>').text('INJ')).appendTo(header);
+				$('<td>').addClass('SmallHead').append($('<div>').text('NI')).appendTo(header);
+				$('<td>').html('<span class ="Team">' + teamName + '</span> P' + nr)
+					.attr('colspan', 9).prop('id','head').appendTo(header);
+				if (mode != 'copyToStatsbook') {
+					$('<td>').addClass('JamTotal').text('JAM').appendTo(header);
+					$('<td>').addClass('GameTotal').text('TOTAL').appendTo(header);
+				}
+			}
+			var body = $('<tbody>').appendTo(table);
+			periodElements[nr] = body;
+			jamElements[nr] = {};
+		}
+	}
+
+
+	function createJam(p, nr) {
+		var table = periodElements[p];
+		if (nr > 0 && jamElements[p][nr] == null) {
+			createJam(p, nr-1);
+
+			var prefix = 'ScoreBoard.Period('+p+').Jam('+nr+').TeamJam('+teamId+').';
+
+			var jamRow = $('<tr>').addClass('Jam').attr('nr', nr);
+			$('<td>').addClass('JamNumber Darker').text(nr).appendTo(jamRow);
+			$('<td>').addClass('Jammer').appendTo(jamRow);
+			$('<td>').addClass('Lost Narrow Darker').click(function() { WS.Set(prefix+'Lost', $(this).text() == ""); }).appendTo(jamRow);
+			$('<td>').addClass('Lead Narrow Darker').click(function() { WS.Set(prefix+'Lead', $(this).text() == ""); }).appendTo(jamRow);
+			$('<td>').addClass('Calloff Narrow Darker').click(function() { WS.Set(prefix+'Calloff', $(this).text() == ""); }).appendTo(jamRow);
+			$('<td>').addClass('Injury Narrow Darker').click(function() { WS.Set(prefix+'Injury', $(this).text() == ""); }).appendTo(jamRow);
+			$('<td>').addClass('NoInitial Narrow Darker').click(function() { setupTripEditor(p, nr, teamId, 1); }).appendTo(jamRow);
+			$.each(new Array(9), function (idx) {
+				var t = idx + 2;
+				$('<td>').addClass('Trip Trip'+t).click(function() { setupTripEditor(p, nr, teamId, t); }).appendTo(jamRow);
 			});
 			if (mode != 'copyToStatsbook') {
-				WS.Register([prefix+'JamScore', prefix+'AfterSPScore'], function(k, v) {
-					jamRow.find('.JamTotal').text(WS.state[prefix+'JamScore'] - WS.state[prefix+'AfterSPScore']);
-				});
-				WS.Register([prefix+'AfterSPScore'], function(k, v) { spRow.find('.JamTotal').text(v)});
-				WS.Register([prefix+'TotalScore', prefix+'AfterSPScore'], function(k, v) {
-					jamRow.find('.GameTotal').text(WS.state[prefix+'TotalScore'] - WS.state[prefix+'AfterSPScore']);
-				});
-				WS.Register([prefix+'TotalScore'], function(k, v) { spRow.find('.GameTotal').text(v)});
+				$('<td>').addClass('JamTotal').appendTo(jamRow);
+				$('<td>').addClass('GameTotal').appendTo(jamRow);
 			}
-			WS.Register([prefix+'Number'], function(k, v) { if(v == null) { jamRow.remove(); spRow.remove(); }})
-			
+
+			var spRow = jamRow.clone(true).removeClass('Jam').addClass('SP Hide');
 			if (mode == 'operator') {
 				table.prepend(jamRow).prepend(spRow);
 			} else {
 				table.append(jamRow).append(spRow);
 			}
+			jamElements[p][nr] = [jamRow, spRow];
+
 		}
 	}
-	
-	function finalizePeriod(nr) {} //TODO: implement
```

Comment:
Why did you remove this? This is still an active todo on my list.

Reply:
It's dead code currently, it can be added back when needed.